### DISCUSSION TITLE
Packaging Angband 4.2.4 for Debian

### DISCUSSION THREAD

**Discussion by magnate** [angband/angband#5622] [General]
Hi all,

I'm back, several years after last updating anything. On my Linux box I found a nice shiny set of packages of angband 4.2.1 which I made two years ago. It took me about half an hour to work out why I hadn't uploaded such nice-looking packages to Debian: the SDL2 port doesn't work - just prints this: 

angband: cant render surface for cache in font '8x13x.fon': Couldn't find glyph

Two years ago I couldn't find anyone who knew how to solve this so I gave up, since it wouldn't have been accepted into Debian without some form of SDL support working. 

Now I've started updating the Debian packaging stuff to 4.2.4, but I haven't yet remembered how to create the Debian source tarball without including the debian/ directory, or how to build in the absence of mk/buildsys.mk (which aborts the package build if it is present). I never did understand why this was so difficult. But I'm really just announcing that I'm here working on this, as there doesn't seem to be another forum to talk to people about it - nobody hangs out in #angband-dev any more!

I will post questions and updates here as I make progress...

**Comment by magnate** [comment]
Update - I have remembered how to build the package (albeit lightyears from a clean source tarball and other requirements). Unfortunately it doesn't run, it gives

angband: Cannot create '/var/games/angband/panic'

... which I assume is some misery related to setgid. But we're underway!

**Comment by backwardsEric** [comment]
In regard to your original post, that issue with SDL2 was due to a change in Freetype 2.11, if I'm remembering correctly.   The related issue for Freetype is here, https://gitlab.freedesktop.org/freetype/freetype/-/issues/1076 , and was fixed in Freetype 2.11.1.

As for the panic directory, that, like the directory for normal save files, should be created during installation for the setgid version.  It should have the same parent directory and permissions as the directory for the normal save files.  I thought that Angband's Makefiles handled its installation, but the packaging for Debian may not install it as it was added between 4.2.3 and 4.2.4.
